java-checks/src/main/java/org/sonar/java/checks/EqualsArgumentTypeCheck.java

```diff
@@ -112,7 +151,11 @@ public void visitBinaryExpression(BinaryExpressionTree tree) {
     @Override
     public void visitMethodInvocation(MethodInvocationTree tree) {
       if (EQUALS_MATCHER.matches(tree)) {
-        typeChecked = true;
+        if (tree.methodSelect().is(Tree.Kind.MEMBER_SELECT) && isArgument(((MemberSelectExpressionTree) tree.methodSelect()).expression())) {
```

Comment:
this line does exactly the same as block 175-178. A method can probably be extracted!